src/blocktridiag.jl

```diff
@@ -0,0 +1,243 @@
+#=======================
+ BlockTridiagMatrix
+=======================#
+
+"""
+    BlockTridiagMatrix{T, R <: AbstractMatrix{T}} <: AbstractBlockArray{T, N}
+
+A `BlockTridiagMatrix` is a block tridiagonal matrix where each block is stored contiguously.
+This means that insertions and retrieval of blocks
+can be very fast and non allocating since no copying of data is needed.
+
+In the type definition, `R` defines the array type that each block has,
+for example `Matrix{Float64}.
+"""
+struct BlockTridiagMatrix{T, R <: AbstractMatrix{T}} <: AbstractBlockMatrix{T}
+    diagl::Vector{R}
```

Comment:
An alternative would be to store the blocks in a `BandedMatrix` from the `BandedMatrices.jl` package, which would then take care of the logic at the block level. But then I don't know if introducing a dependency on `BandedMatrices.jl` is a good idea.